### PR TITLE
add git blame api

### PR DESCRIPTION
**Pull request by rebornix**
add git blame api to git extension, right now I just use string for the result and users take care of the parsing.

### CODE REVIEW

File: extensions/git/src/git.ts [hunk 1]
```diff
@@ -1282,6 +1282,19 @@ export class Repository {
...
+	async blame(path: string): Promise<string> {
+		try {
+			const args = ['blame'];
+			args.push(path);
+
+			let result = await this.run(args);
+
+			return result.stdout.trim();
+		} catch (err) {
+			throw err;
```
Comment (joaomoreno) [inline extensions/git/src/git.ts]: Leftovers? :smile:

File: extensions/git/src/repository.ts [hunk 2]
```diff
@@ -999,6 +1000,10 @@ export class Repository implements Disposable {
 		await this.run(Operation.Push, () => this.repository.push(remote, undefined, false, true, forcePushMode));
 	}
 
+	async blame(path: string): Promise<string> {
+		return await this.run(Operation.Blame, () => this.repository.blame(path));
```
Comment (joaomoreno) [inline extensions/git/src/repository.ts]: Should we parse the blame result into something more structured?

**Review by joaomoreno — CHANGES_REQUESTED** [review]

File: extensions/git/src/git.ts (hunk 1 above)
Comment (rebornix) [reply extensions/git/src/git.ts]: Nice catch, for now we only need to specificily handle one error type `no such path`.

File: extensions/git/src/repository.ts (hunk 2 above)
Comment (rebornix) [reply extensions/git/src/repository.ts]: Similar to `git log`, the output of `git blame` is more or less *free form* once we add more options to it. I thought about returing `string[]` but for `--porcelain`, one info entry for a line can be multiline so I'd like to suggest to leave the parsing work to callers. Does this make sense?

**Pull request merged**